src/hooks/useCurrencyConverter.js

```diff
@@ -0,0 +1,40 @@
+import { useState, useEffect } from "react";
+
+function calculateExchangeRate(currencies, fromCurrency, toCurrency) {
+  if (!currencies || !currencies.rates) {
+    return;
+  }
+  if (fromCurrency === currencies.base) {
```

Comment:
Can you explain more about your choice of having a series if else here ? Can we use case or other methods somehow ?

Reply:
The checking is a bit messy here, which I admit. The purpose is to first check if the currency object is defined (and the rates inside) and return if it's undefined, then check if the fromCurrency is the same as the base currency (which is EUR in the api, btw) and if so, return the exchange rate for the toCurrency from the object, and then check if toCurrency is equal to the base currency, in which case it returns the inverse of the exchange rate for fromCurrency. Finally if neither fromCurrency or toCurrency is the base currency, the function will calculate the exchange rate between the two currencies by dividing their exchange rates.

I'm all about new ideas to simplify this conversion, might be worth it to explore a switch case or another way to make it more graceful